packages/mask/src/plugins/CryptoartAI/utils.ts

```diff
@@ -0,0 +1,71 @@
+import BigNumber from 'bignumber.js'
+import type { AssetEvent, Order } from 'opensea-js/lib/types'
+import { parseURL } from '../../utils/utils'
+import { ChainId, formatBalance } from '@masknet/web3-shared-evm'
+import { pathnameRegexMatcher, mainNetwork, testNetwork } from './constants'
+import type { CryptoartAIToken } from './types'
+
+export function checkUrl(url: string): boolean {
+    const protocol = 'https://'
+    const _url = new URL(url.startsWith(protocol) ? url : protocol + url)
+    return (
+        [mainNetwork.hostname, testNetwork.hostname].includes(_url.hostname) &&
+        _url.pathname.includes(pathnameRegexMatcher)
+    )
+}
+
+export function getRelevantUrl(textContent: string) {
+    const urls = parseURL(textContent)
+    return urls.find(checkUrl)
+}
+
+export function getAssetInfoFromURL(url?: string) {
+    if (!url) return null
+    const _url = new URL(url)
+    const urls = _url.pathname
+        .substring(_url.pathname.indexOf(pathnameRegexMatcher) + pathnameRegexMatcher.length)
+        .split('/')
+    if (urls.length >= 2) {
+        return {
+            chain_id: url.includes(testNetwork.hostname) ? ChainId.Kovan : ChainId.Mainnet,
+            creator: urls[0],
+            token_id: urls[1],
+            contractAddress: url.includes(testNetwork.hostname)
+                ? testNetwork.contractAddress
+                : mainNetwork.contractAddress,
+        }
+    }
+
+    return
```

Comment:
```ts
const addresses = {
    [ChainId.Kovan]: testNetwork.contractAddress,
    [ChainId.Mainnet]: mainNetwork.contractAddress,
}


const { hostname, pathname } = new URL(url)
const pattern = new RegExp(`^${pathnameRegexMatcher}\\/([^\\/]+)/([^\\/]+)$`)
const matched = pattern.exec(pathname)
if (!matched) {
    return // early return
}

const chain_id = hostname === testNetwork.hostname ? ChainId.Kovan : ChainId.Mainnet
const creator = matched[1]
const token_id = matched[2]
const contract_address = addresses[chain_id]
return { chain_id, creator, token_id, contract_address }
```

the example code need test.

Reply:
I think, `pathnameRegexMatcher` constant need rename.

e.g: `galleryPath`, `prefixPath` or others.